Mathlib/LinearAlgebra/UnitaryGroup.lean

```diff
@@ -198,6 +198,21 @@ set_option linter.uppercaseLean3 false in
 
 end UnitaryGroup
 
+section specialUnitaryGroup
+
+variable (n) (α : Type v) [CommRing α] [StarRing α] {A : Matrix n n α}
+
+/--`Matrix.specialUnitaryGroup` is the group of unitary `n` by `n` matrices where the determinant
+is 1. (This definition is only correct if 2 is invertible.)-/
+abbrev specialUnitaryGroup := unitaryGroup n α ⊓ MonoidHom.mker detMonoidHom
+
+theorem mem_specialUnitaryGroup_iff  (h : A ∈ unitaryGroup n α):
+    A ∈ specialUnitaryGroup n α ↔ A.det = 1 := by
```

Comment:
I would expect a lemma with this name to say that `A` is special unitary if and only if `A` is unitary and has determinant `1`. Then you could add a followup lemma for the special case where `A` is already known to be unitary (maybe called `unitary_mem_specialUnitaryGroup_iff`).

Reply:
Ok, the follow up lemma could be the subject for another PR....